### PR TITLE
fix: Delegated Accounts Linting

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 432762c</samp>

Improved code quality and readability in `delegated-accounts.page.ts` by adding type annotations and removing unused imports. This file handles the logic for switching between delegated and own accounts in the app.

### <samp>🤖 Generated by Copilot at 432762c</samp>

> _`Delegated-accounts` of doom, we clean up your code_
> _No unused imports, we follow the mode_
> _Type annotations, we make them explicit_
> _Readability and quality, we make them exquisite_

### <samp>🤖 Generated by Copilot at 432762c</samp>

*  Specify types for element references, parameters, return values, variables and properties to improve type safety and clarity ([link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL27-R19), [link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL44-R36), [link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL61-R59), [link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL90-R86), [link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL98-R95))
* Reorder and remove imports to enhance readability and maintainability ([link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL2-R11))
* Use `fromEvent` observable to listen to input changes and filter delegatees by name ([link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL90-R86))
* Implement `switchToDelegatee` method to switch the current user to the selected delegatee and navigate to the dashboard ([link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL44-R36))
* Implement `switchToOwn` method to switch back to the original user and reload the app ([link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL61-R59))
* Fetch and display the list of delegatees from the `orgUserService` in the `ngOnInit` and `ionViewWillEnter` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL61-R59), [link](https://github.com/fylein/fyle-mobile-app/pull/2429/files?diff=unified&w=0#diff-06c6e53d083bff9e0b7a86ad57db1f892d20ef04b0e7f282764f23c5b3acc59bL98-R95))

## Clickup
https://app.clickup.com/t/85ztznwc3

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/delegated-accounts/delegated-accounts.page.ts [hunk 1]
```diff
@@ -1,30 +1,22 @@
 import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
-import { Router, ActivatedRoute } from '@angular/router';
-import { forkJoin, from, fromEvent, throwError } from 'rxjs';
-import { OrgUserService } from 'src/app/core/services/org-user.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { User } from '@sentry/angular';
```
Comment (Chethan-Fyle) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.ts:3]: Are we using `User` from sentry library?

File: src/app/fyle/delegated-accounts/delegated-accounts.page.ts [hunk 2]
```diff
@@ -87,18 +81,18 @@ export class DelegatedAccountsPage implements OnInit {
...
           switchMap((searchText) =>
             delegatedAccList$.pipe(
               map(({ delegatedAcc }) => this.orgUserService.excludeByStatus(delegatedAcc, 'DISABLED')),
-              map((delegatees) =>
-                delegatees?.filter((delegatee) =>
+              map((delegatees: ExtendedOrgUser[]) =>
+                delegatees?.filter((delegatee: ExtendedOrgUser) =>
                   Object.values(delegatee.us).some(
-                    (delegateeProp) =>
+                    (delegateeProp: User) =>
```
Comment (Chethan-Fyle) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.ts:95]: The `User` being used here is from sentry library. Is this intentional?

**Review by Chethan-Fyle — CHANGES_REQUESTED** [review]
One doubt.

File: src/app/fyle/delegated-accounts/delegated-accounts.page.ts [hunk 3]
```diff
@@ -58,11 +50,13 @@ export class DelegatedAccountsPage implements OnInit {
       });
   }
 
-  ngOnInit() {}
+  ngOnInit(): void {
```
Comment (OmkarJ13) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.ts:53]: Why not remove this entirely?

File: src/app/fyle/delegated-accounts/delegated-accounts.page.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/delegated-accounts/delegated-accounts.page.ts]: It has was imported by mistake, have added the correct type.

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**